src/Components/Navbar/Navbar.js

```diff
@@ -0,0 +1,137 @@
+import "./navbar.css";
+import ShoppingCartIcon from "@mui/icons-material/ShoppingCart";
+import FavoriteIcon from "@mui/icons-material/Favorite";
+import { Link } from "react-router-dom";
+import { useProducts } from "./../../Context/products-context";
+import { useNavbar } from "./../../Context/navbar-context";
+import { useEffect } from "react";
+
+function Navbar() {
+  const encodedToken = localStorage.getItem("ENCODED_TOKEN");
+
+  const { isAuthenticated, setIsAuthenticated } = useNavbar();
+
+  useEffect(() => {
+    if (encodedToken !== null) {
+      setIsAuthenticated(true);
+      setNavbarButtonText("Logout");
+    }
+  }, []);
+
+  const { cartArray, setCartArray, setWishlistArray, wishlistArray } =
+    useProducts();
+
+  const { navbarButtonText, setNavbarButtonText } = useNavbar();
+
+  const handleNavbar = () => {
+    const items = document.querySelectorAll(".nav-items li");
+    const burger = document.querySelector(".burger");
+    const nav = document.querySelector(".nav-items");
+
+    nav.classList.toggle("navSlide");
+    items.forEach((item) => {
+      item.style.animation = "";
+    });
+    burger.classList.toggle("toggle");
+  };
```

Comment:
Using JS to directly manipulate variables is an anti pattern for React JS
Prefer using controlled states for handling such behaviour

Reply:
I have fixed this by using a State hook. Thanks for pointing it out, Shrey.